infection_monkey/windows_upgrader.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import os
+import struct
+import subprocess
+import sys
+
+import time
+
+import monkeyfs
+from config import WormConfiguration
+from control import ControlClient
+from exploit.tools import build_monkey_commandline_explicitly
+from model import MONKEY_CMDLINE_WINDOWS
+
+__author__ = 'itay.mizeretz'
+
+LOG = logging.getLogger(__name__)
+
+if "win32" == sys.platform:
+    from win32process import DETACHED_PROCESS
+else:
+    DETACHED_PROCESS = 0
+
+
+class WindowsUpgrader(object):
+    @staticmethod
+    def is_64bit_os():
+        return os.environ.has_key('PROGRAMFILES(X86)')
+
+    @staticmethod
+    def is_64bit_python():
+        return struct.calcsize("P") == 8
+
+    @staticmethod
+    def is_windows_os():
+        return sys.platform.startswith("win")
+
+    @staticmethod
+    def should_upgrade():
+        return WindowsUpgrader.is_windows_os() and WindowsUpgrader.is_64bit_os() \
+               and not WindowsUpgrader.is_64bit_python()
+
+    @staticmethod
+    def upgrade(opts):
+        monkey_64_path = ControlClient.download_monkey_exe_by_os(True, False)
+        with monkeyfs.open(monkey_64_path, "rb") as downloaded_monkey_file:
+            monkey_bin = downloaded_monkey_file.read()
+        with open(WormConfiguration.dropper_target_path_win_64, 'wb') as written_monkey_file:
+            written_monkey_file.write(monkey_bin)
+
+        depth = int(opts.depth) if opts.depth is not None else None
```

Comment:
I'd rather not cast here, but add a parameter for the -d parser in monkey.initialize to cast it to int.

Reply:
fixed